src/reboost/math/functions.py

```diff
@@ -58,10 +59,85 @@ def piecewise_linear_activeness(
     else:
         distances_ak = distances
 
+    dl = fccd * dlf
+
     # compute the linear piecewise
     results = ak.where(
         (distances_ak > fccd) | np.isnan(distances_ak),
         1,
-        ak.where(distances_ak <= tl, 0, (distances_ak - tl) / (fccd - tl)),
+        ak.where(distances_ak <= dl, 0, (distances_ak - dl) / (fccd - dl)),
     )
     return VectorOfVectors(results) if results.ndim > 1 else Array(results.to_numpy())
+
+
+def vectorised_active_energy(
+    distances: VectorOfVectors | ak.Array,
+    edep: VectorOfVectors | ak.Array,
+    fccd: float | list,
+    dlf: float | list,
+) -> VectorOfVectors | Array:
+    r"""Energy after piecewise linear HPGe activeness model vectorised over FCCD or dead layer fraction.
+
+    Based on the same linear activeness function as :func:`piecewise_linear_activeness`. However,
+    this function vectorises the calculation to provide a range of output energies varying the fccd or
+    dead layer fraction. Either fccd or dlf can be a list. This adds an extra dimension to the
+    output.
+
+    > *Warning*
+    > It is not currently implemented to vary both dlf and fccd.
```

Comment:
```suggestion
    .. warning:
        It is not currently implemented to vary both dlf and fccd.
```

the old syntax does not work